driver-sync/src/main/com/mongodb/client/internal/Crypt.java

```diff
@@ -348,7 +350,9 @@ private void decryptKeys(final MongoCryptContext cryptContext) {
             }
             cryptContext.completeKeyDecryptors();
         } catch (Throwable t) {
-            throw wrapInClientException(t);
+            throw translateInterruptedException(t, "Interrupted while reading")
```

Comment:
Is it necessary to translate interrupted exceptions both here and in the KeyManagementService (which this is calling)?

Reply:
Only if we want to add messages like "interrupted while connecting/writing/reading" which is what we do for streams, and why I did it this way here. But if it's fine to just have a single message "interrupted while doing IO", or no message at all, I'll be happy to remove the translating code from `Crypt`.

Reply:
Done.